weave-js/src/components/PagePanelComponents/Home/Browse3/pages/common/KeyValueTable.tsx

```diff
@@ -0,0 +1,209 @@
+import {Box} from '@mui/material';
+import _ from 'lodash';
+import moment from 'moment';
+import React, {useEffect, useState} from 'react';
+
+import {parseRefMaybe, SmallRef} from '../../../Browse2/SmallRef';
+
+const VALUE_SPACE = 4;
+const ROW_HEIGHT = 26;
+const PADDING_GAP = 16;
+const MAX_HEIGHT_MULT = 5;
+
+export const KeyValueTable: React.FC<{
+  data: {[key: string]: any};
+  headerTitle?: string;
+}> = props => {
+  return (
+    <Box
+      sx={{
+        border: '1px solid rgba(224, 224, 224, 1)',
+        borderRadius: '4px',
+      }}>
+      <table
+        style={{
+          width: '100%',
+          borderCollapse: 'collapse',
+          tableLayout: 'fixed',
+        }}>
+        {props.headerTitle && (
+          <thead>
+            <tr
+              style={{
+                borderBottom: '1px solid rgba(224, 224, 224, 1)',
+                backgroundColor: '#FAFAFA',
+              }}>
+              <th colSpan={VALUE_SPACE + 1}>{props.headerTitle}</th>
+            </tr>
+          </thead>
+        )}
+        <tbody>
+          <KeyValueRowForObject objValue={props.data} />
+        </tbody>
+      </table>
+    </Box>
+  );
+};
+
+const baseKeyStyle: React.CSSProperties = {
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+
+  padding: '4px, 8px, 4px, 8px',
+
+  fontWeight: 600,
+
+  height: `${ROW_HEIGHT}px`,
+};
+
+const parentKeyStyle: React.CSSProperties = {
+  ...baseKeyStyle,
+  borderTop: '1px solid rgba(224, 224, 224, 1)',
+  borderBottom: '1px solid rgba(224, 224, 224, 1)',
+  backgroundColor: '#FAFAFA',
+  color: '#979a9e',
+  position: 'sticky',
+  top: 0,
+  zIndex: 1,
+};
+
+const leafKeyStyle: React.CSSProperties = {
+  ...baseKeyStyle,
+  textAlign: 'right',
+  borderRight: '1px solid rgba(224, 224, 224, 1)',
+  paddingRight: '8px',
+  verticalAlign: 'top',
+  backgroundColor: '#fff',
+  borderBottom: '1px solid rgba(224, 224, 224, 1)',
+};
+
+const valueStyle: React.CSSProperties = {
+  paddingLeft: '8px',
+  verticalAlign: 'top',
+  borderBottom: '1px solid rgba(224, 224, 224, 1)',
+};
+
+const KeyValueRow: React.FC<{
+  rowKey: string;
+  rowValue: any;
+}> = props => {
+  const depth = React.useContext(DepthContext);
```

Comment:
I'd get rid of `React.` and just add to import list as you already have for `useState`

Reply:
Done